### DISCUSSION TITLE
Are you using the new `workspace:` protocol feature?

### DISCUSSION THREAD

**Discussion by ghiscoding** [ghiscoding/lerna-lite#156] [Polls]
Great news, Lerna-Lite just brought support of yarn/pnpm `workspace:` protocol, are you using it? Please take 30sec. to complete this poll and perhaps share your project. 

Thanks for using Lerna-Lite and happy coding 🎉

**Comment by meteorlxy** [comment]
Removing other commands of lerna is great, because most of us (as I know) only need a tool for version bumping and publishing. Changesets & Rush (as recommended in pnpm docs)  are too heavy. Original lerna does not support `workspace:` protocol, and we only want `lerna version` and `lerna publish` commands.

I finally chose to use following toolchain with pnpm workspaces before knowing lerna-lite:

- `bumpp` for version bumping (have to bump all packages in the same time, even not modified)
- `conventional-changelog-cli` for changelog (could not generate changelog for every single packages)
- `pnpm -r publish` for publishing

As you can see, it's just a simplified alternative of `lerna version` and `lerna publish`.

So, I thinks it would be better to make lerna-lite even more light-weighted and focusing on version bumping and publishing. For example, making `init` and `info` commands optional, too. Or, allowing the `@lerna-lite/publish` package to be installed and used separately.

BTW, as all package managers has supported workspaces, we can consider removing `packages` and `useWorkspaces` config. Respecting the value of `npmClient` should be enough. Also, if we are not going to remove them, `useWorkspaces` currently does not work with `pnpm-workspace.yaml`.

**Reply by ghiscoding** [reply]
Thanks for the feedback, however I am not intending to change anything. I agree that `version` and `publish` are what most people will want to use (myself included) and that is why I've put them in the core and CLI, I also added `init` and `info` in the built-in CLI because they are extremely small and also because I wanted users to get started in an easy way with `init` while `info` is helpful when opening a new issue and again they are extremely small packages. All other commands will be optional, but anyway at the end of the day this is all Dev dependencies so it doesn't matter much. 

As for `useWorkspaces`, it follows Lerna's code implementation, it was introduced when Yarn Workspaces came out they use and declare the `workspaces` property in the `package.json` instead of `lerna.json` (and now npm as well but not pnpm) hence the name `useWorkspaces` (pnpm follows a different approach and you can't use that flag but plenty of other users do use it and need it). To be clear, I intend to follow the same code logic as Lerna and not change any flags/options that was already introduced by them, I also don't intend to remove anything, it's easier to replicate any new PRs coming in Lerna and replicate into Lerna-Lite and Lerna-Lite is meant to be a drop-in replacement (except for optional commands which require an extra install). Your suggestion to remove `packages` and `useWorkspaces` unfortunately doesn't stand because these options help Lerna find where the packages are located, so you're using pnpm, great, but that's not the case for everyone else (a lot of people are still using yarn 1.x classic) and removing flags like this would most certainly break someone's project and I will not risk that to just remove 10 lines of code.

I would suggest to open other [Discussions](https://github.com/ghiscoding/lerna-lite/discussions) to discuss off topic subjects, this post is outside of the poll's survey. Thanks

**Reply by meteorlxy** [reply]
Yeah, I understand your are making it a direct replacement of lerna. What I posted was just some simple ideas, and might be reasonable enough for this project.

**Comment by ghiscoding** [comment]
Starting with release [v1.5.0](https://github.com/ghiscoding/lerna-lite/releases/tag/v1.5.0), we now have a new flag [`--sync-workspace-lock`](https://github.com/ghiscoding/lerna-lite/tree/main/packages/version#--sync-workspace-lock) to help updating the lock file, it's taking full advantage of your package manager client to update the lock (make sure to configure `npmClient` in your `lerna.json` config file). This new flag is opt-in and so you will need to enable it yourself, it can be very useful with pnpm/yarn3.6+ workspaces but won't work with yarn 1.x classic which is why it's opt-in

**Comment by njfamirm** [comment]
yarn v3.6
